RTEMS5: Make QEMU tests work

mdavidsaver opened the pull request:
Partially address #175.

Rewrite the DHCP handler to avoid repeated explicit `char*` handling.

Make `rtemsInit_NTP_server_ip` optional, and remove the default.

Change the dhcpcd.conf option name separator from `_` to `-`.  May be necessary.

mdavidsaver committed:
RTEMS5: redo dhcp handler and make NTP optional

mdavidsaver commented:
@hjunkes @brendanchandler fyi.

hjunkes commented:
Looks ok to me. Thanks
As for the names in dhcpcd.cnf with "_" or "-" I am unsure. In the documentation of FreeBSD is usually worked with "_". 
https://www.freebsd.org/cgi/man.cgi?query=dhcpcd&sektion=8&apropos=0&manpath=FreeBSD+11.0-RELEASE+and+Ports

brendanchandler commented:
Looks reasonable to me.

mdavidsaver committed:
RTEMS5: update libbsd logging

Show messages synchronously during boot,
then redirect through errlog before user app.

Disable syslog() during tests

mdavidsaver committed:
RTEMS: e1000 giant hack for QEMU w/ libbsd

mdavidsaver commented:
I've made some progress towards running RTEMS5 test with QEMU.  I still haven't been able to make `-net nic,model=rtl8139` work for me (with QEMU >=3.1.0).  However, I found an ugly hack to make `-net nic,model=e1000` boot faster by stubbing out the eeprom checksum verification code.  With this change the trivial `epicsTypeTest` will run for me in 14 seconds (vs. 4 seconds with RTEMS 4.x).  The remaining time seems to be mostly taken by 3x duplicate address checks.

```
sending ARP probe (1 of 3), next in %0.1f seconds
```

(yes, the printf spec is being printed without expansion :) )

Adding 10 seconds to every test still more than doubles the total runtime, but it is at least in the realm of the practical for me.

I've also added a change to re-enable logging from libbsd (boy do I wish I'd noticed this sooner...).  The change leaves the default handler (w/ synchronous `printf()`) in place until near the end of `POSIX_Init()`, then sets a handler using `errlogVprintf()`.

mdavidsaver committed:
RTEMS: rtems_netconfig.c version test consistency

The comment above says "... no longer needed in RTEMS 4.11"
which to me says needed "< 4.11".

mdavidsaver committed:
ci: run tests for RTEMS 4.9 and 5
